src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_linux.go

```diff
@@ -714,6 +729,45 @@ func saveDmidecode(ctx context.Context, ps *stepParams) step {
 	}
 }
 
+func saveControllerLogDir(cmdFs afero.Fs, ps *stepParams, cfg *config.Config, logLimitBytes int) step {
+	return func() error {
+		controllerDir := filepath.Join(cfg.Redpanda.Directory, "redpanda", "controller", "0_0")
+
+		// We don't need the .base_index files to parse out the messages.
+		exclude := regexp.MustCompile(`^*.base_index`)
+		size, err := osutil.DirSize(controllerDir, exclude)
+		if err != nil {
+			return err
+		}
+
+		if int(size) > logLimitBytes {
+			// TODO: This will be changed to a custom truncation of the logs.
+			return fmt.Errorf("controller logs not stored, size is too big (%v). You can adjust the limit by changing --controller-logs-size-limit flag", units.HumanSize(float64(size)))
+		}
+
+		// Create a temporary directory to store the filtered files and add them
+		// to the zip writer.
+		tmpDir, err := afero.TempDir(cmdFs, "", "controller-")
+		if err != nil {
+			return err
+		}
+		defer cmdFs.RemoveAll(tmpDir)
+
+		copyOpts := cp.Options{
+			Skip: func(_ os.FileInfo, src, _ string) (bool, error) {
+				return exclude.MatchString(src), nil
+			},
+			PreserveTimes: true,
+		}
+		err = cp.Copy(controllerDir, tmpDir, copyOpts)
+		if err != nil {
+			return err
+		}
+
+		return writeDirToZip(ps, tmpDir, "controller")
```

Comment:
Rather than having a two step path, can we make this one step?

I think this first copies all files needed to a new tmp dir, then copies that new tmp dir to the bundle, then deletes the new tmp dir. Can it be a single pass, where we copy directly the files needed into the bundle without a tmp dir?